### PR TITLE
Feature/notifications

### PR DESCRIPTION
**Pull request by ProgramistaJarek**

### CODE REVIEW

File: src/app/favorites/data-access/api-service/favorites.store.ts [hunk 1]
```diff
@@ -7,165 +7,167 @@ import { ApiFavoritesService } from './api-favorites.service';
...
+    const newImage: Favorites[] = state.cats.map((img) => {
+      if (img.image_id === id.image_id) {
+        this.api.setNotifications({
+          name: 'Zagłosowano pozytywnie',
+          date: Date.now(),
+          id: img.id,
+          type: 'favourites',
+          action: 'like',
+        });
+        console.log({ ...img, vote: 1 });
```
Comment (TeapotCode) [inline src/app/favorites/data-access/api-service/favorites.store.ts:52]: 52: console.log

File: src/app/favorites/data-access/api-service/favorites.store.ts [hunk 2]
```diff
@@ -7,165 +7,167 @@ import { ApiFavoritesService } from './api-favorites.service';
...
+    const newImage: Favorites[] = state.cats.map((img) => {
+      if (img.image_id === id.image_id) {
+        this.api.setNotifications({
+          name: 'Zagłosowano negatywnie',
+          date: Date.now(),
+          id: img.id,
+          type: 'favourites',
+          action: 'unlike',
+        });
+        console.log({ ...img, vote: -1 });
```
Comment (TeapotCode) [inline src/app/favorites/data-access/api-service/favorites.store.ts:76]: 76: console.log

File: src/app/favorites/data-access/api-service/favorites.store.ts (hunk 1 above)
Comment (ProgramistaJarek) [reply src/app/favorites/data-access/api-service/favorites.store.ts:52]: te consol logi to nie moje tylko Zuzi

File: src/app/favorites/data-access/api-service/favorites.store.ts (hunk 2 above)
Comment (ProgramistaJarek) [reply src/app/favorites/data-access/api-service/favorites.store.ts:76]: i ten

File: src/app/shell/data-access/cats.reducer.ts [hunk 3]
```diff
@@ -44,5 +47,13 @@ export const catsReducer = createReducer(
       favoriteImages: newFavImages,
       voteImages: newVoteImages,
     };
+  }),
+
+  on(catsAction.setNotification, (state, { notifi }) => {
+    console.log('dziala', notifi);
```
Comment (TeapotCode) [inline src/app/shell/data-access/cats.reducer.ts:53]: 52: console.log

File: src/app/votes/data-access/votes.store.ts [hunk 4]
```diff
@@ -33,9 +33,20 @@ export class VotesStore extends ComponentStore<VotesState> {
   }));
 
   readonly updateImagesIfDislike = this.updater((state, id: number) => {
-    const newImage: voteImages[] = state.image.map((img) =>
-      img.favoriteId === id ? { ...img, isFavorite: false } : img
-    );
+    const newImage: voteImages[] = state.image.map((img) => {
+      if (img.favoriteId === id) {
+        this.api.setNotifications({
+          name: 'Usunięto z ulubionych',
```
Comment (TeapotCode) [inline src/app/votes/data-access/votes.store.ts:39]: 39: to chyba po angielsku powinno być

File: src/app/votes/data-access/votes.store.ts [hunk 5]
```diff
@@ -44,11 +55,20 @@ export class VotesStore extends ComponentStore<VotesState> {
...
   readonly updateImagesIfLike = this.updater((state, id: helperIfLike) => {
-    const newImage: voteImages[] = state.image.map((img) =>
-      img.image_id === id.id
-        ? { ...img, isFavorite: true, favoriteId: id.result.id }
-        : img
-    );
+    const newImage: voteImages[] = state.image.map((img) => {
+      if (img.image_id === id.id) {
+        this.api.setNotifications({
+          name: 'Dodano do ulubionych',
```
Comment (TeapotCode) [inline src/app/votes/data-access/votes.store.ts:61]: 61: tak samo

File: src/app/shell/ui/navbar/navbar.component.scss [hunk 6]
```diff
@@ -8,3 +8,32 @@ mat-toolbar {
...
+  user-select: none;
+  cursor: pointer;
+  outline: 0;
+  border: none;
+  -webkit-tap-highlight-color: transparent;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: block;
+  /** set line height to desired height */
```
Comment (tomryb) [inline src/app/shell/ui/navbar/navbar.component.scss:25]: delete comment for production

**Review by TeapotCode — COMMENTED** [review]
tldr: console.log

File: src/app/favorites/data-access/api-service/favorites.store.ts (hunk 1 above)
Comment (TeapotCode) [reply src/app/favorites/data-access/api-service/favorites.store.ts:52]: ok

File: src/app/favorites/data-access/api-service/favorites.store.ts [hunk 7]
```diff
@@ -7,165 +7,167 @@ import { ApiFavoritesService } from './api-favorites.service';
...
+      )
+    );
+  });
+
+  readonly delete = this.effect((cats$: Observable<toVote>) => {
+    return cats$.pipe(
+      mergeMap((cat) =>
+        this.api.removeVote(cat.voteId).pipe(
+          tap(() => this.ifRemoveVote(cat))
+          //{
```
Comment (tomryb) [inline src/app/favorites/data-access/api-service/favorites.store.ts:165]: delete commented code if it's unused

**Review by tomryb — CHANGES_REQUESTED** [review]
